gnovm/pkg/gnolang/values.go

```diff
@@ -273,7 +273,35 @@ func (pv PointerValue) Assign2(alloc *Allocator, store Store, rlm *Realm, tv2 Ty
 		// Special case of DataByte into (base=*SliceValue).Data.
 		pv.TV.SetDataByte(tv2.GetUint8())
 		return
+	} else if rlm != nil && pv.Base == nil && pv.Index == 0 {
+		// An index of zero indicates this is an assignment to a realm
```

Comment:
An index of zero could mean it's the zero'th element of an array, or the zero'th element of a block, 
or whatever the .Base type is. The Index doesn't say much but the .Base type says more but depends on what you mean by a realm object. Like a PackageValue?

Reply:
You're right that an index of zero can mean more than one thing depending on the context, but in this case the realm is not nil and the base is nil, so I think this means that a value is being assigned to a realm object. I'm not sure what the difference is between realm object and PackageValue; they might be synonymous -- the values that are persisted from one block to the next; variables that are declared in a realm and are not local to any function or method.

Reply:
ok at least the comment needs fixing, "A base of nil indicates"? 
A base of nil indicates an assignment to an escaped object.
An escaped object has no owner.
All objects are realm objects.
Some are substructures (base == parent, ref==1), others are escaped.
Unless I'm confused haha. 

All realms have an associated package value but not all packages are realms. Depends on the PkgPath name.